test/test_algos.py

```diff
@@ -14,97 +15,69 @@
 
 try:
     import torch
+    from lensless import UnrolledFISTA, UnrolledADMM
 
-    torch_vals = [True, False]
+    torch_is_available = True
+    torch.autograd.set_detect_anomaly(True)
+    trainable_algos = [UnrolledFISTA, UnrolledADMM]
 except ImportError:
-    torch_vals = [False]
+    torch_is_available = False
+    trainable_algos = []
 
 
 psf_fp = "data/psf/tape_rgb.png"
 data_fp = "data/raw_data/thumbs_up_rgb.png"
 dtypes = ["float32", "float64"]
 downsample = 16
-n_iter = 5
+_n_iter = 5
 
+# classical algorithms
+standard_algos = [GradientDescent, NesterovGradientDescent, FISTA, ADMM]
 
-def test_gradient_descent():
-    for gray in [True, False]:
-        for dtype in dtypes:
-            for torch_data in torch_vals:
-                psf, data = load_data(
-                    psf_fp=psf_fp,
-                    data_fp=data_fp,
-                    downsample=downsample,
-                    plot=False,
-                    gray=gray,
-                    dtype=dtype,
-                    torch=torch_data,
-                )
-                recon = GradientDescent(psf, dtype=dtype)
-                recon.set_data(data)
-                res = recon.apply(n_iter=n_iter, disp_iter=None, plot=False)
-                assert psf.shape[3] == (1 if gray else 3)
-                assert res.dtype == psf.dtype, f"Got {res.dtype}, expected {dtype}"
-
-
-def test_admm():
-    for gray in [True, False]:
-        for dtype in dtypes:
-            for torch_data in torch_vals:
-                psf, data = load_data(
-                    psf_fp=psf_fp,
-                    data_fp=data_fp,
-                    downsample=downsample,
-                    plot=False,
-                    gray=gray,
-                    dtype=dtype,
-                    torch=torch_data,
-                )
-                recon = ADMM(psf, dtype=dtype)
-                recon.set_data(data)
-                res = recon.apply(n_iter=n_iter, disp_iter=None, plot=False)
-                assert psf.shape[3] == (1 if gray else 3)
-                assert res.dtype == psf.dtype, f"Got {res.dtype}, expected {dtype}"
-
-
-def test_nesterov_gradient_descent():
+
+@pytest.mark.parametrize("algorithm", standard_algos)
```

Comment:
This creates a unique test per algorithm? I.e. a unique "." when running `pytest`?

Reply:
Yes, this seems to be the "proper" way to use pytest

Reply:
Good to know. I agree it looks nicer